pkg/api/types.go

```diff
@@ -518,8 +518,18 @@ type GitRepoVolumeSource struct {
 // The contents of the target Secret's Data field will be presented in a volume
 // as files using the keys in the Data field as the file names.
 type SecretVolumeSource struct {
-	// Name of the secret in the pod's namespace to use
-	SecretName string `json:"secretName"`
+	// Name of the secret in the pod's namespace to use.  May not be specified if items is specified.
+	SecretName string `json:"secretName,omitempty"`
+	// A list of projections to make into the volume.  May not be specified if SecretName is specified.
+	Items []SecretVolumeFile `json:"items,omitempty"`
+}
+
+// SecretVolumeFile represents a single secret key projected into a file.
+type SecretVolumeFile struct {
+	// A reference to the secret key to project into the volume.
+	SecretKeySelector `json:",inline"`
```

Comment:
@pmorie We cannot change fields from required to optional or vice versa, since the former breaks readers and the latter breaks writers. We have fixed the specification of omitempty to match the existing behavior a few times, but that's different than changing the observed behavior.